source/github-events/on-new-comments.ts

```diff
@@ -4,7 +4,17 @@ import delegate from 'delegate-it';
 const discussionsWithListeners = new WeakSet();
 const handlers = new Set<VoidFunction>();
 const delegates = new Set<delegate.Subscription>();
-const observer = new MutationObserver(run);
+
+let commentsCount = 0;
+const observer = new MutationObserver(() => {
+	const commentsNewCount = select.all('.js-comment').length;
```

Comment:
Why not move this inside the `run` function?

Reply:
Because the other delegates calling `run` can't be triggered by unrelated events.

That said, it might be worth keeping the comment count up-to-date in `run()`, to handle the following situation:
1. The user clicks on "Load more" to load hidden comments
2. This is handled by the first delegate so `commentsCount` isn't updated.
3. Later, the MutationObserver is triggered by a random event (e.g. adding a label) and sees the comment count has changed, so it calls `run` unnecessarily.